### DISCUSSION TITLE
Blogpost 6

### DISCUSSION THREAD

**Discussion by Hpixpoke** [LeonRother/SE_Platzverwaltung#7] [General]
Hallo zusammen,

vergangene Woche haben wir uns ans Login gemacht. Vorab: es ist noch nicht fertig. Wir wollen eigentlich JWT und BCrypt benutzen, hatten aber ein paar Schwierigkeiten:
- BCrypt wird i.d.R. im Backend verwendet. Dafür muss das Password aber im Klartext übertragen werden. Das ist eigentlich schlecht, wird in der Praxis aber durch HTTP**S** mitigiert. Weil wir aber nur lokal entwickeln, haben wir nur HTTP zur Verfügung. Meine Idee war ursprünglich, das Passwort im Frontend zu hashen und dann zu übertragen, das ist aber mit BCrypt nicht möglich, weil der Salt jedes mal unterschiedlich wäre. Man bräuchte entweder salt-free Algorithmen (z.B. SHA512, die aber suboptimal bzgl. Kollisionen sind), oder man überträgt halt Klartext. Wir versuchen's nochmal richtig im Backend zu machen, wechseln aber notfalls auf eine einfachere Lösung, falls das zu viel Zeit beanspruchen wird.
- JWT ist theoretisch garnicht so schwer, in der Implementierung aber etwas tricky, wenn mans noch nie gemacht hat. Wir wollen es mal versuchen, richtig zu machen und dieses "Sandbox"-Projekt als Lernmöglichkeit zu nutzen, es gilt aber selbes wie oben.

Sonst sind neue Tabellen `Employees` und `Organizations` entstanden. Hier noch ein Screenshot von Login-Screen:

![image](https://github.com/user-attachments/assets/2e368ad7-401a-4fd4-b4e5-7d022476187f)

Außerdem heißt unser Produkt nun _Deskflex_. Wenn jemand weiß, wie man Vue mit HTTPS startet, gerne kommentieren ;) Mich würde auch interessieren, wie ihr eure Login-Logik gestaltet.

Viele Grüße
Hagen & Team Deskflex

**Comment by philipsagawe** [comment]
Hallo Hagen & Team Deskflex,

spannender Einblick in eure Arbeit, danke dafür!

Zu JWT: Achtet darauf, nur nicht-sensible Daten in den Token-Payload zu schreiben. Es lohnt sich, hier etwas mehr Zeit für die Implementierung zu investieren.
Für HTTPS mit Vue: In der vite.config.js könnt ihr mit mkcert generierte Zertifikate einbinden. Das könnte euch schnell weiterhelfen.

Viele Grüße
Philip

**Comment by SafaeKa** [comment]
Hi,
Die UI sieht erstmal gut aus. Wir haben noch keine login Funktion in Green Sprout entwickelt, von daher kann ich noch nichts vergleichen. Habt ihr es vor eure Anwendung am ende zu deployen? Dann ist das HTTP erstmal egal auf der lokalen Ebene, und könnte erst später gelöst werden. Sonst kann man eigentlich HTTPS lokal benutzen, wenn man kannst ein selbstsigniertes SSL-Zertifikat erstellt und dieses in dem lokalen Server integriert. Ich kenne mich leider nicht so gut damit aus aber vielleicht gibt es Anleitungen dazu online.

Wir wünschen euch viel Erfolg!
GaLiGrü 😊
Paula, Safae, Samuel, Valentin, Jonas
